### PR TITLE
Authorisation scheme

### PR DESCRIPTION
**Pull request by Vegter**

### CODE REVIEW

File: src/gobapi/auth/auth_query.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        return [attr for attr, auth in self.get_checked_columns().items()
+                if not any([role for role in roles if role in auth.get('roles', [])])]
+
+    def filter_row(self, row):
+        """
+        Set all columns in the row that should be suppressed to None
+        """
+        suppressed_columns = self.get_suppressed_columns()
+        if suppressed_columns:
+            for column in suppressed_columns:
```
Comment (JKoetsier) [inline src/gobapi/auth/auth_query.py]: if suppressed_columns can be left out, then simplify to `for column in self.get_suppressed_columns()` ?

File: src/gobapi/auth/auth_query.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            if suppressed_columns:
+                self.set_suppressed_columns(entity, suppressed_columns)
+                for column in suppressed_columns:
+                    self.suppress_attribute(column, entity)
+            yield entity
+
+    def suppress_attribute(self, column, entity):
+        try:
+            # For query results, delattr sets the entity value to null
+            delattr(entity, column)
```
Comment (JKoetsier) [inline src/gobapi/auth/auth_query.py]: setattr(entity, column, None)

File: src/gobapi/auth/auth_query.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        roles = self.get_roles()
+        return [attr for attr, auth in self.get_checked_columns().items()
+                if not any([role for role in roles if role in auth.get('roles', [])])]
+
+    def filter_row(self, row):
+        """
+        Set all columns in the row that should be suppressed to None
+        """
+        suppressed_columns = self.get_suppressed_columns()
+        for column in [c for c in suppressed_columns if row.get(c) is not None]:
```
Comment (JKoetsier) [inline src/gobapi/auth/auth_query.py]: if c in row

File: src/gobapi/graphql/filters.py [hunk 4]
```diff
@@ -90,6 +97,22 @@ def resolve_attribute(obj, info, **kwargs):
     return resolve_attribute
 
 
+def get_resolve_auth_attribute(catalog_name, collection_name, attr, resolver):
```
Comment (JKoetsier) [inline src/gobapi/graphql/filters.py]: Check if necessary

File: src/gobapi/auth/auth_query.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+class AuthorizedQuery(Query):
+
+    def __init__(self, *args, **kwargs):
+        """
+        An authorized query checks every entity for columns that should not be communicated.
+        """
+        self._authority = None
+        super().__init__(*args, **kwargs)
+
+    def set_catalog_collection(self, catalog, collection):
```
Comment (JKoetsier) [inline src/gobapi/auth/auth_query.py]: See if it possible to derive this from the entity argument in the constructor, as discussed.

File: src/gobapi/auth/auth_query.py (hunk 2 above)
Comment (Vegter) [reply src/gobapi/auth/auth_query.py]: Fixed

File: src/gobapi/auth/auth_query.py (hunk 3 above)
Comment (Vegter) [reply src/gobapi/auth/auth_query.py]: Fixed

File: src/gobapi/graphql/filters.py (hunk 4 above)
Comment (Vegter) [reply src/gobapi/graphql/filters.py]: Removed, good point!

File: src/gobapi/auth/auth_query.py (hunk 5 above)
Comment (Vegter) [reply src/gobapi/auth/auth_query.py]: Well, this is not the case...

**Pull request merged**